projects/storefrontlib/src/cms-structure/page/slot/page-slot.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PageSlotService {
+  protected instantSsrSlots: string[] | undefined;
+
+  constructor(
+    @Inject(PLATFORM_ID) platformId: any,
+    @Inject(DOCUMENT) document
+  ) {
+    if (isPlatformBrowser(platformId)) {
```

Comment:
Let's avoid logic in the constructor to not block extensibily.

Reply:
👍